server/datastore/mysql/migrations/tables/20210818151827_RemoveForeignKeysSchedQStats.go

```diff
@@ -0,0 +1,46 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20210818151827, Down_20210818151827)
+}
+
+func Up_20210818151827(tx *sql.Tx) error {
+	rows, err := tx.Query(`SELECT DISTINCT CONSTRAINT_NAME, REFERENCED_TABLE_NAME FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE WHERE TABLE_NAME = 'scheduled_query_stats' AND CONSTRAINT_NAME <> 'PRIMARY'`)
+	if err != nil {
+		return errors.Wrap(err, "getting fk for scheduled_query_stats")
+	}
+	var constraints []string
+	for rows.Next() {
+		var constraintName string
+		var referencedTable string
+		err := rows.Scan(&constraintName, &referencedTable)
+		if err != nil {
+			return errors.Wrap(err, "scanning fk for scheduled_query_stats")
+		}
+		if referencedTable == "hosts" || referencedTable == "scheduled_queries" {
+			constraints = append(constraints, constraintName)
+		}
+	}
+	if len(constraints) == 0 {
+		return errors.New("Found no constraints in scheduled_query_stats")
+	}
+
+	for _, constraint := range constraints {
+		_, err = tx.Exec(fmt.Sprintf(`ALTER TABLE scheduled_query_stats DROP FOREIGN KEY %s;`, constraint))
+		if err != nil {
+			return errors.Wrapf(err, "dropping fk %s", constraint)
+		}
+	}
+	return nil
+}
+
+func Down_20210818151827(tx *sql.Tx) error {
+	return nil
```

Comment:
I assume we use migrations in a "forward-only" mode, otherwise we'd want the "Down" to be the reverse the "Up" precisely (i.e. in this case recreate the dropped constraints)? That's more for my personal understanding.

Reply:
Correct. The reasoning behind it would be a question for @zwass though.

Reply:
That's right. Over the first couple of years of Fleet, we never found an occasion where the down migrations were used (we wrote them but never exposed a command to run them). My belief is that due to the lossy nature of many migrations (dropping columns, etc.), it's more advisable to fail forward (by releasing a new version with further migrations that fix the issue) or revert to a DB backup than try to run a down migration. 

I'm open to discussion on this.

Reply:
I strongly agree with forward-only migrations, I think that down migrations can cause subtle issues (e.g. an "up" migration that removes an index on a big table, and the "down" migration just re-creates the index, taking a long time to complete, causing disk space issues, etc.). I think forward-only forces you to think through what's going to happen when applying the migration as you write it when it's needed, while the down migration may have been written a long time ago (with very low expectations of ever running). That's how I've implemented migrations (fwd only) in a couple of web frameworks I've written.